server/bin/pbench-unpack-tarballs.py

```diff
@@ -72,11 +78,11 @@ def unpack_tarball(
                 "Tarball link, '{}', does not resolve to a real location", tarball
             )
 
-        controller_path = tb.parent
-        controller = Controller.create(controller_path.name, config, logger)
-        tar = controller.create_tarball(tb)
+        tar_md5 = get_tarball_md5(tb)
+        tar = file_tree.find_dataset(tar_md5)
+
         try:
-            tar.unpack(incoming, results)
+            tar.controller.unpack(incoming, results)
         except Exception:
```

Comment:
Ah; indeed, I missed the fact that the `Controller` level `unpack` is used to make sure the controller's INCOMING and RESULTS directories exist.

However, we can compact this and make it look a bit better. Turns out that I actually wrote a `FileTree.unpack(<resource_id>)`, so this would be neater as just:

```
    try:
        file_tree.unpack(get_tarball_md5(tb))
    except Exception:
```